content/webapp/pages/exhibition-guide.tsx

```diff
@@ -196,6 +207,34 @@ export const getServerSideProps: GetServerSideProps<Props | AppErrorProps> =
       ),
     };
 
+    const userPreferenceGuideType = getCookie('WC_userPreferenceGuideType', {
+      req,
+      res,
+    });
+    const hasUserPreference = hasCookie('WC_userPreferenceGuideType', {
+      req,
+      res,
+    });
+
+    // We want to set a user preference cookie if the qr code url contains a guide type
+    // 28000 is 8 hours - the average length of time the collection is open for
+    if (usingQRCode) {
+      setCookie('WC_userPreferenceGuideType', type, {
+        res,
+        req,
+        maxAge: 28800,
```

Comment:
The number in the comment doesn't match the value. I'd suggest ditching a hard-coded value and replacing it with something that makes the meaning easier to deduce, i.e.

```suggestion
        maxAge: 8 * 60 * 60,
```

Separately:

* Let's put the cookie variable name in a constant
* Why aren't you using the `cookieHandler` function you've added?
* Is it the average length of time (per the comment), or the longest amount of time the galleries are open for a day (per the PR description)? Let's be accurate.

Reply:
The cookieHandler function is used on the client side, using cookies-next in server side means we need to provide both res and req in the options object.

Reply:
> The number in the comment doesn't match the value. I'd suggest ditching a hard-coded value and replacing it with something that makes the meaning easier to deduce, i.e.
> 
> Separately:
> 
> * Let's put the cookie variable name in a constant
> * Why aren't you using the `cookieHandler` function you've added?
> * Is it the average length of time (per the comment), or the longest amount of time the galleries are open for a day (per the PR description)? Let's be accurate.

Agreed, accuracy is important here. I’ll adjust

Reply:
> The cookieHandler function is used on the client side, using cookies-next in server side means we need to provide both res and req in the options object

Ah, of course – I hadn't spotted the client/server-side distinction. Thanks for explaning!